kivy/core/window/_window_sdl2.pyx

```diff
@@ -686,3 +707,36 @@ cdef SDL_Surface* flipVert(SDL_Surface* sfc):
         memcpy(&rpixels[pos], &pixels[(pxlength - pos) - pitch], pitch)
 
     return result
+
+cdef SDL_Surface* surface_from_core_image(core_image):
+    # Create an SDL_Surface from a core image's texture pixels. Use
+    # SDL_FreeSurface(surface) to deallocate.
+    #
+    # Returns NULL if the core image has no texture pixels, or if
+    # there's a problem with the creation of an SDL Surface.
+    cdef int width, height
+    width, height = core_image.size
+    cdef int pitch = width * 4
+    cdef bytes pixels
+    cdef char* sdl_error
+    try:
+        pixels = core_image.image.texture.pixels
+    except AttributeError:
+        return NULL
+
+    if pixels == None:
+        return NULL
+
+    cdef Uint8 *c_pixels = pixels
+    cdef SDL_Surface *surface
+    surface = SDL_CreateRGBSurface(0, width, height, 32,
+        0x00000000ff, 0x0000ff00, 0x00ff0000, 0xff000000)
+    if surface == NULL:
+        sdl_error = SDL_GetError()
+        Logger.warn(
+            "Window: Can't create a surface from a core image: %s" % sdl_error
+        )
+        return NULL
+
+    memcpy(surface.pixels, c_pixels, pitch * height)
+    return surface
```

Comment:
This looks like it leaks memory? Is there a problem using try/finally in this situation?

```python
try:
    return surface
finally:
    SDL_FreeSurface(surface)
```

Reply:
Indeed this leaks memory, as mentioned in my [earlier comment](https://github.com/kivy/kivy/pull/5785#issuecomment-395694274). But it's not possible to just call `SDL_FreeSurface()` there. Or at least I was unable to make that work IIRC.